modules/core/src/lib/deck.ts

```diff
@@ -375,21 +375,32 @@ export default class Deck<ViewsT extends ViewOrViews = null> {
     // See if we already have a device
     if (props.device) {
       this.device = props.device;
-    } else if (props.gl) {
+    }
+
+    let deviceOrPromise: Device | Promise<Device> | null = this.device;
+
+    // Attach a new luma.gl device to a WebGL2 context if supplied
+    if (!deviceOrPromise && props.gl) {
       if (props.gl instanceof WebGLRenderingContext) {
         log.error('WebGL1 context not supported.')();
       }
-      this.device = WebGLDevice.attach(props.gl);
+      deviceOrPromise = webgl2Adapter.attach(props.gl);
     }
 
-    let deviceOrPromise: Device | Promise<Device> | null = this.device;
+    // Create a new device
     if (!deviceOrPromise) {
-      // TODO v9 should we install WebGL backend as default for now?
-      luma.registerDevices([WebGLDevice]);
-
+      // Create the "best" device supported from the registered adapters
       deviceOrPromise = luma.createDevice({
+        type: 'best-available',
+        adapters: [webgl2Adapter],
         ...props.deviceProps,
-        canvas: this._createCanvas(props)
+        createCanvasContext: {canvas: this._createCanvas(props)},
+        _resourceDefaults: {
```

Comment:
@felixpalmer Added a mechanism to let us override resource default props. Trying to replicate the improvements you saw.

Reply:
@ibgreen the [implementation](https://github.com/visgl/luma.gl/blob/29af54969e04c8bf31376e9b03bca7e242d2afb6/modules/core/src/adapter/resources/texture.ts#L420) is confusing for me, rather than providing defaults it overrides any passed in props. I think this is too aggressive to be useful